File: src/tests/DataBlock Sample Application/VB.NET/Orders/OrdersTableMetadata.vb
Imports System
Imports System.Data
Imports voidsoft.DataBlock
Namespace ExtenderOrders

    Public Class OrderDetailsTableMetadata
        Inherits TableMetadata
        Private _fields As DatabaseField()

        Public Sub New()
            _fields = New DatabaseField(4) {}
            _fields(0) = New DatabaseField(DbType.Int32, "OrderID", False, False, Nothing)
            _fields(1) = New DatabaseField(DbType.Int32, "ProductID", False, False, Nothing)
            _fields(2) = New DatabaseField(DbType.Decimal, "UnitPrice", False, False, Nothing)
            _fields(3) = New DatabaseField(DbType.Int16, "Quantity", False, False, Nothing)
            _fields(4) = New DatabaseField(DbType.Single, "Discount", False, False, Nothing)
            Me.currentTableName = "Order Details"
        End Sub

        Public Overloads Overrides Property TableFields() As DatabaseField()
            Get
                Return _fields
            End Get
            Set(ByVal Value As DatabaseField())
                _fields = value
            End Set
        End Property

        Public Property OrderID() As System.Int32
            Get
                Return CType((Me.GetField("OrderID")).fieldValue, System.Int32)
            End Get
            Set(ByVal Value As System.Int32)
                Me.SetFieldValue("OrderID", value)
            End Set
        End Property

        Public Property ProductID() As System.Int32
            Get
                Return CType((Me.GetField("ProductID")).fieldValue, System.Int32)
            End Get
            Set(ByVal Value As System.Int32)
                Me.SetFieldValue("ProductID", value)
            End Set
        End Property

        Public Property UnitPrice() As System.Decimal
            Get
                Return CType((Me.GetField("UnitPrice")).fieldValue, System.Decimal)
            End Get
            Set(ByVal Value As System.Decimal)
                Me.SetFieldValue("UnitPrice", value)
            End Set
        End Property

        Public Property Quantity() As System.Int16
            Get
                Return CType((Me.GetField("Quantity")).fieldValue, System.Int16)
            End Get
            Set(ByVal Value As System.Int16)
                Me.SetFieldValue("Quantity", value)
            End Set
        End Property

        Public Property Discount() As System.Single
            Get
                Return CType((Me.GetField("Discount")).fieldValue, System.Single)
            End Get
            Set(ByVal Value As System.Single)
                Me.SetFieldValue("Discount", value)
            End Set
        End Property
    End Class

    Public Class OrdersTableMetadata
        Inherits TableMetadata
        Private _fields As DatabaseField()

        Public Sub New()
            _fields = New DatabaseField(13) {}
            _fields(0) = New DatabaseField(DbType.Int32, "OrderID", True, True, Nothing)
            _fields(1) = New DatabaseField(DbType.String, "CustomerID", False, False, Nothing)
            _fields(2) = New DatabaseField(DbType.Int32, "EmployeeID", False, False, Nothing)
            _fields(3) = New DatabaseField(DbType.DateTime, "OrderDate", False, False, Nothing)
            _fields(4) = New DatabaseField(DbType.DateTime, "RequiredDate", False, False, Nothing)
            _fields(5) = New DatabaseField(DbType.DateTime, "ShippedDate", False, False, Nothing)
            _fields(6) = New DatabaseField(DbType.Int32, "ShipVia", False, False, Nothing)
            _fields(7) = New DatabaseField(DbType.Decimal, "Freight", False, False, Nothing)
            _fields(8) = New DatabaseField(DbType.String, "ShipName", False, False, Nothing)
            _fields(9) = New DatabaseField(DbType.String, "ShipAddress", False, False, Nothing)
            _fields(10) = New DatabaseField(DbType.String, "ShipCity", False, False, Nothing)
            _fields(11) = New DatabaseField(DbType.String, "ShipRegion", False, False, Nothing)
            _fields(12) = New DatabaseField(DbType.String, "ShipPostalCode", False, False, Nothing)
            _fields(13) = New DatabaseField(DbType.String, "ShipCountry", False, False, Nothing)
            Me.tableName = "Orders"
            Me.alRelations.Add(New TableRelation("FK_OrderDetails", "Order Details", "OrderID"))
        End Sub

        Public Overloads Overrides Property TableFields() As DatabaseField()
            Get
                Return _fields
            End Get
            Set(ByVal Value As DatabaseField())
                _fields = value
            End Set
        End Property

        Public Property OrderID() As System.Int32
            Get
                Return CType((Me.GetField("OrderID")).fieldValue, System.Int32)
            End Get
            Set(ByVal Value As System.Int32)
                Me.SetFieldValue("OrderID", value)
            End Set
        End Property

        Public Property CustomerID() As System.String
            Get
                Return CType((Me.GetField("CustomerID")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("CustomerID", value)
            End Set
        End Property

        Public Property EmployeeID() As System.Int32
            Get
                Return CType((Me.GetField("EmployeeID")).fieldValue, System.Int32)
            End Get
            Set(ByVal Value As System.Int32)
                Me.SetFieldValue("EmployeeID", value)
            End Set
        End Property

        Public Property OrderDate() As Date
            Get
                Return CType((Me.GetField("OrderDate")).fieldValue, Date)
            End Get
            Set(ByVal Value As Date)
                Me.SetFieldValue("OrderDate", value)
            End Set
        End Property

        Public Property RequiredDate() As Date
            Get
                Return CType((Me.GetField("RequiredDate")).fieldValue, Date)
            End Get
            Set(ByVal Value As Date)
                Me.SetFieldValue("RequiredDate", value)
            End Set
        End Property

        Public Property ShippedDate() As Date
            Get
                Return CType((Me.GetField("ShippedDate")).fieldValue, Date)
            End Get
            Set(ByVal Value As Date)
                Me.SetFieldValue("ShippedDate", value)
            End Set
        End Property

        Public Property ShipVia() As System.Int32
            Get
                Return CType((Me.GetField("ShipVia")).fieldValue, System.Int32)
            End Get
            Set(ByVal Value As System.Int32)
                Me.SetFieldValue("ShipVia", value)
            End Set
        End Property

        Public Property Freight() As System.Decimal
            Get
                Return CType((Me.GetField("Freight")).fieldValue, System.Decimal)
            End Get
            Set(ByVal Value As System.Decimal)
                Me.SetFieldValue("Freight", value)
            End Set
        End Property

        Public Property ShipName() As System.String
            Get
                Return CType((Me.GetField("ShipName")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("ShipName", value)
            End Set
        End Property

        Public Property ShipAddress() As System.String
            Get
                Return CType((Me.GetField("ShipAddress")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("ShipAddress", value)
            End Set
        End Property

        Public Property ShipCity() As System.String
            Get
                Return CType((Me.GetField("ShipCity")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("ShipCity", value)
            End Set
        End Property

        Public Property ShipRegion() As System.String
            Get
                Return CType((Me.GetField("ShipRegion")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("ShipRegion", value)
            End Set
        End Property

        Public Property ShipPostalCode() As System.String
            Get
                Return CType((Me.GetField("ShipPostalCode")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("ShipPostalCode", value)
            End Set
        End Property

        Public Property ShipCountry() As System.String
            Get
                Return CType((Me.GetField("ShipCountry")).fieldValue, System.String)
            End Get
            Set(ByVal Value As System.String)
                Me.SetFieldValue("ShipCountry", value)
            End Set
        End Property
    End Class
End Namespace

File: src/tests/DataBlock Sample Application/CSharp/Orders/OrdersTableMetadata.cs
using System;
using System.Data;
using voidsoft.DataBlock;

namespace ExtenderOrders
{


	public class OrderDetailsTableMetadata : TableMetadata
	{
		private DatabaseField[] _fields;

		public OrderDetailsTableMetadata()
		{
			_fields = new DatabaseField[5];
			_fields[0] = new DatabaseField(DbType.Int32,"OrderID",false,false,null);
			_fields[1] = new DatabaseField(DbType.Int32,"ProductID",false,false,null);
			_fields[2] = new DatabaseField(DbType.Decimal,"UnitPrice",false,false,null);
			_fields[3] = new DatabaseField(DbType.Int16,"Quantity",false,false,null);
			_fields[4] = new DatabaseField(DbType.Single,"Discount",false,false,null);
 
			this.currentTableName = "Order Details";


		}


		public override DatabaseField[] TableFields 
		{
			get{ return _fields;}
			set{_fields = value;}
		}

		public System.Int32 OrderID
		{
			get
			{
				return (System.Int32) (this.GetField("OrderID")).fieldValue;
			}

			set
			{
				this.SetFieldValue("OrderID", value);
			}
		}


		public System.Int32 ProductID
		{
			get
			{
				return (System.Int32) (this.GetField("ProductID")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ProductID", value);
			}
		}


		public System.Decimal UnitPrice
		{
			get
			{
				return (System.Decimal) (this.GetField("UnitPrice")).fieldValue;
			}

			set
			{
				this.SetFieldValue("UnitPrice", value);
			}
		}


		public System.Int16 Quantity
		{
			get
			{
				return (System.Int16) (this.GetField("Quantity")).fieldValue;
			}

			set
			{
				this.SetFieldValue("Quantity", value);
			}
		}


		public System.Single Discount
		{
			get
			{
				return (System.Single) (this.GetField("Discount")).fieldValue;
			}

			set
			{
				this.SetFieldValue("Discount", value);
			}
		}

	}


	public class OrdersTableMetadata : TableMetadata
	{
		private DatabaseField[] _fields;

		public OrdersTableMetadata()
		{
			_fields = new DatabaseField[14];
			_fields[0] = new DatabaseField(DbType.Int32,"OrderID",true,true,null);
			_fields[1] = new DatabaseField(DbType.String,"CustomerID",false,false,null);
			_fields[2] = new DatabaseField(DbType.Int32,"EmployeeID",false,false,null);
			_fields[3] = new DatabaseField(DbType.DateTime,"OrderDate",false,false,null);
			_fields[4] = new DatabaseField(DbType.DateTime,"RequiredDate",false,false,null);
			_fields[5] = new DatabaseField(DbType.DateTime,"ShippedDate",false,false,null);
			_fields[6] = new DatabaseField(DbType.Int32,"ShipVia",false,false,null);
			_fields[7] = new DatabaseField(DbType.Decimal,"Freight",false,false,null);
			_fields[8] = new DatabaseField(DbType.String,"ShipName",false,false,null);
			_fields[9] = new DatabaseField(DbType.String,"ShipAddress",false,false,null);
			_fields[10] = new DatabaseField(DbType.String,"ShipCity",false,false,null);
			_fields[11] = new DatabaseField(DbType.String,"ShipRegion",false,false,null);
			_fields[12] = new DatabaseField(DbType.String,"ShipPostalCode",false,false,null);
			_fields[13] = new DatabaseField(DbType.String,"ShipCountry",false,false,null);
 
			this.currentTableName = "Orders";

			this.alRelations.Add(new TableRelation("FK_OrderDetails", "Order Details", "OrderID"));


		}


		public override DatabaseField[] TableFields 
		{
			get{ return _fields;}
			set{_fields = value;}
		}

		public System.Int32 OrderID
		{
			get
			{
				return (System.Int32) (this.GetField("OrderID")).fieldValue;
			}

			set
			{
				this.SetFieldValue("OrderID", value);
			}
		}


		public System.String CustomerID
		{
			get
			{
				return (System.String) (this.GetField("CustomerID")).fieldValue;
			}

			set
			{
				this.SetFieldValue("CustomerID", value);
			}
		}


		public System.Int32 EmployeeID
		{
			get
			{
				return (System.Int32) (this.GetField("EmployeeID")).fieldValue;
			}

			set
			{
				this.SetFieldValue("EmployeeID", value);
			}
		}


		public System.DateTime OrderDate
		{
			get
			{
				return (System.DateTime) (this.GetField("OrderDate")).fieldValue;
			}

			set
			{
				this.SetFieldValue("OrderDate", value);
			}
		}


		public System.DateTime RequiredDate
		{
			get
			{
				return (System.DateTime) (this.GetField("RequiredDate")).fieldValue;
			}

			set
			{
				this.SetFieldValue("RequiredDate", value);
			}
		}


		public System.DateTime ShippedDate
		{
			get
			{
				return (System.DateTime) (this.GetField("ShippedDate")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ShippedDate", value);
			}
		}


		public System.Int32 ShipVia
		{
			get
			{
				return (System.Int32) (this.GetField("ShipVia")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ShipVia", value);
			}
		}


		public System.Decimal Freight
		{
			get
			{
				return (System.Decimal) (this.GetField("Freight")).fieldValue;
			}

			set
			{
				this.SetFieldValue("Freight", value);
			}
		}


		public System.String ShipName
		{
			get
			{
				return (System.String) (this.GetField("ShipName")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ShipName", value);
			}
		}


		public System.String ShipAddress
		{
			get
			{
				return (System.String) (this.GetField("ShipAddress")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ShipAddress", value);
			}
		}


		public System.String ShipCity
		{
			get
			{
				return (System.String) (this.GetField("ShipCity")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ShipCity", value);
			}
		}


		public System.String ShipRegion
		{
			get
			{
				return (System.String) (this.GetField("ShipRegion")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ShipRegion", value);
			}
		}


		public System.String ShipPostalCode
		{
			get
			{
				return (System.String) (this.GetField("ShipPostalCode")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ShipPostalCode", value);
			}
		}


		public System.String ShipCountry
		{
			get
			{
				return (System.String) (this.GetField("ShipCountry")).fieldValue;
			}

			set
			{
				this.SetFieldValue("ShipCountry", value);
			}
		}

	}
}
